tests/python/pants_test/base/test_cmd_line_spec_parser.py

```diff
@@ -76,6 +76,10 @@ def test_globs(self) -> None:
     for glob in ["*", "**/*", "a/b/*", "a/b/test_*.py", "a/b/**/test_*"]:
       self.assert_filesystem_spec_parsed(glob)
 
+  def test_excludes(self) -> None:
+    for glob in ["!", "!a/b/", "!/a/b/*"]:
```

Comment:
Would we ever want to consider using `!` for target specs as well?

Reply:
I don't think so? Generally, the address specs API is very stable and I see no reason we would expand it. 

I also expect that most people will end up defaulting to filesystem specs thanks to the convenience shell auto-complete.

Reply:
> I also expect that most people will end up defaulting to filesystem specs thanks to the convenience shell auto-complete.

Would you like to sync to discuss intelligent and performant tab-completion for pants targets on the command line? I think this might be doable with a week of work by one person.

Reply:
That'd be neat! However, our priority at Toolchain is entirely on filesystem specs and the prework necessary to make that work well (hence this change). We likely won't even teach address specs to our users expect as an advanced concept. 

Pants and Bazel are the only tools I know of that require you to work at the target level when on the CLI. Most users are more familiar with file arguments because that's how things like NPM, Pytest, Black, Prettier, Scalafmt, and Junit work.